sdk/template/azure-template/CMakeLists.txt

```diff
@@ -54,6 +54,12 @@ set(
 )
 
 add_library(azure-template ${AZURE_TEMPLATE_HEADER} ${AZURE_TEMPLATE_SOURCE})
+
+# SDK libraries should privately define _azure_BUILDING_SDK,
+# tests should define _azure_BUILDING_TESTS,
+# samples should have _azure_BUILDING_SAMPLES.
+target_compile_definitions(azure-template PRIVATE _azure_BUILDING_SDK)
```

Comment:
Thanks for adding the comment.

This is missing the rationale for **why** we need to add these different definitions in each `CMakeLists.txt`.

Reply:
As I said, "we will be able to answer the questions about the amount of telemetry coming from SDK, tests, samples, and customer code". That is the intent. Customers are not expected to experiment with these values or rationalize them. I documented what they should be.

Reply:
> we will be able to answer the questions about the amount of telemetry coming from SDK, tests, samples, and customer code

Anton, let's be more concrete here. What are example questions we will get an answer for by doing this?

Why do we care, at all, how much telemetry data is coming from our test and samples? I don't see any reason for why we need this distinction.

Reply:
FWIW, I can absolutely see a value in distinguishing between our live tests and our live samples. For instance, we can (hopefully) tell if customers are building our samples and running them or if they are just being run from our CI pipelines.